commatrix/generate.go

```diff
@@ -0,0 +1,175 @@
+package commatrix
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+
+	"golang.org/x/sync/errgroup"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clientutil "github.com/openshift-kni/commatrix/client"
+	"github.com/openshift-kni/commatrix/consts"
+	"github.com/openshift-kni/commatrix/debug"
+	"github.com/openshift-kni/commatrix/ss"
+	"github.com/openshift-kni/commatrix/types"
+)
+
+// generate relevant matrix and diff files
+func GenerateCommatrix(kubeconfig, customEntriesPath, customEntriesFormat, format string, env Env, deployment Deployment, printFn func(m types.ComMatrix) ([]byte, error), destDir string) {
+	mat, err := New(kubeconfig, customEntriesPath, customEntriesFormat, env, deployment)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create the communication matrix: %s", err))
```

Comment:
We should not panic but rather return the err and let the caller decide what to do with it (same for the other panics)